app/src/main/java/com/example/android/weatherapp/fragment/HomeScreenFragment.kt

```diff
@@ -9,77 +9,27 @@ import androidx.lifecycle.Observer
 import androidx.lifecycle.ViewModelProviders
 import com.example.android.weatherapp.viewmodel.HomeScreenViewModel
 import com.example.android.weatherapp.R
+import com.example.android.weatherapp.adapter.HomeScreenAdapter
 import com.example.android.weatherapp.databinding.FragmentHomeScreenBinding
+import com.example.android.weatherapp.models.CurrentWeatherModel
 
 class HomeScreenFragment : BaseFragment() {
+    val viewModel: HomeScreenViewModel by lazy {
+        ViewModelProviders.of(this).get(HomeScreenViewModel::class.java)
+    }
 
     override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
         savedInstanceState: Bundle?): View? {
-        val binding: FragmentHomeScreenBinding = DataBindingUtil.inflate(
-            inflater, R.layout.fragment_home_screen, container, false)
-
-       /* val adapter = HomeScreenAdapter()
-        binding.homeScreenRcv.adapter = adapter*/
-
-
-        val viewModel: HomeScreenViewModel by lazy {
-            ViewModelProviders.of(this).get(HomeScreenViewModel::class.java)
-        }
-
+        val binding: FragmentHomeScreenBinding = DataBindingUtil.inflate(inflater,
+            R.layout.fragment_home_screen, container, false)
 
        viewModel.currentWeather.observe(viewLifecycleOwner, Observer { currentData ->
             currentData?.let {
```

Comment:
Make a separate method to observe LiveData.

Reply:
Created separate method in pull#4